pkg/envoy/lds/response.go

```diff
@@ -44,6 +33,10 @@ func NewResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_d
 
 	lb := newListenerBuilder(meshCatalog, svcAccount.ToServiceIdentity(), cfg, statsHeaders)
 
+	if cfg.GetFeatureFlags().EnableMulticlusterMode && proxy.Kind() == envoy.KindGateway {
```

Comment:
I'd move the multicluster check inside buildGatewayListeners, considering that function needs to build the listeners for ingress.